res/schema.xml

```diff
@@ -446,4 +446,37 @@ METADATA
       );
     </sql>
   </revision>
+  <revision version="29" min_compatible="3">
+    <description>
+      Add tables to support nested crates.
+    </description>
+    <sql>
+      CREATE TABLE IF NOT EXISTS crateClosure (
```

Comment:
Whith AUTOINCREMENT SQLite needs to keep track of the greatest ID value that has ever been used in a separate table. Without the keyword it can simply leverage the primary key index of the table itself and reuse any currently unused ID value:
https://www.sqlite.org/autoinc.html

Two use cases for AUTOCOMMIT:
* if you need to store the ID somewhere else outside of the database and you must guarantee that an ID still identifies the same record when coming back later at any time
* if you are using the ID to enforce a strict ordering of records, i.e. for a persistent FIFO task scheduler